File: 2341-maximum-number-of-pairs-in-array/2341-maximum-number-of-pairs-in-array.kt
class Solution {
    fun numberOfPairs(nums: IntArray): IntArray {
        val arr = IntArray(101){0}
        val res = IntArray(2){0}
        for (n in nums) {
            if (arr[n] !== 0) {
                arr[n] = 0
                res[0]++
                res[1]--
            } else {
                arr[n] = 1
                res[1]++
            }
        }
        return res
    }
}

File: 2341-maximum-number-of-pairs-in-array/2341-maximum-number-of-pairs-in-array.py
class Solution:
    def numberOfPairs(self, nums: List[int]) -> List[int]:
        arr = [0] * 101
        res = [0, 0]
        for n in nums:
            if arr[n] != 0:
                arr[n] = 0
                res[0] += 1
                res[1] -= 1
            else:
                arr[n] = 1
                res[1] += 1
        return res